### ISSUE TITLE
View FBX files exported from Autodesk Navisworks with threejs

### ISSUE THREAD

**Issue by ColPC** [mrdoob/three.js#16720]
Hi there

we are investigating the use of ThreeJS to display 3D models exported from Autodesk Navisworks.
We exported a test model from Navisworks and tried to view it using ThreeJS.
We experimented with the FBX export formats.
We’ve had success with FBX (after resaving it with Paint 3D).

We used FBXLoader. When trying to open the FBX file as exported form Navisworks, it fails to display. The error from the loader is: TypeError: Cannot read property ‘replace’ of undefined.

The console logs looks like this (I added logging the output of the loader):

THREE.WebGLRenderer 105
THREE.FBXLoader: FBX binary version: 7500
TypeError: Cannot read property ‘replace’ of undefined

When I however open the FBX file (as exported from Navisworks) with Paint 3D and save a copy of it as an FBX file, it displays correctly in ThreeJS .

console logs:
THREE.WebGLRenderer 105
THREE.FBXLoader: FBX binary version: 7400

Lewy had a look at it for us and confirmed that the FBXLoader were not able to parse the FBX file. 

Lewy requested me to open a new issue for this.

Originally submitted on discourse:
https://discourse.threejs.org/t/view-fbx-kmz-files-exported-from-autodesk-navisworks/7963/4

I attach the FBX file that was exported from Navisworks for your perusal.

Thank you
[stair.zip](https://github.com/mrdoob/three.js/files/3264510/stair.zip)

##### Browser

- [x] All of them
- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [x] All of them
- [ ] Windows
- [ ] macOS
- [ ] Linux
- [ ] Android
- [ ] iOS

**Comment by seabeams** [comment]
@ColPC Did you ever resolve this issue?

**Comment by looeee** [comment]
This model now loads. 

I'm not sure which PR fixed this. I missed that it was working since its offset by a huge amount from the origin and is at a very large scale. 

`object.position.set(-41011106, -156210024, -818221)` brings it into view. 

There's still a 90 degree error in rotation though. 

![2020-04-16 17_42_31-three js webgl - FBX loader](https://user-images.githubusercontent.com/5307958/79447398-00393980-800a-11ea-8691-8f11b636920e.png)

Child meshes also have `userData` specifying this transform:

![userdata](https://user-images.githubusercontent.com/5307958/79448146-1abfe280-800b-11ea-8104-4becce025912.png)

I think its common to specify extra transforms like this in Autodesk BIM software, since I've encountered something similar on Revit models before. 

This is not something I intend to investigate further at this time. Users working with Autodesk BIM models should traverse the scene and manually apply these transforms if they want to bring the model into the scene center.

**Comment by looeee** [comment]
[According to this document](https://help.autodesk.com/view/NAVFREE/2021/ENU/?guid=GUID-698C9B45-4C49-4368-8EE6-F4FA47E79B2B), the order of axes in Navisworks can be edited by the user. 

I've also tested this model in Sketchfab and they don't get the rotation right either. 

Given that, I don't think we'll be able to fix the rotation error. This will have to be handled by users along with the user data transform. 

As a result, and since the original problem has been fixed, I think it's OK to close this issue.
